---
title: "main_figures"
output: html_document
---

 Generates all main figure panels for the manuscript 
 (except Fig 1a, generated by basic_oncoprint.Rmd and Fig 3, generated by therapy_timelines.R)
    Output should go in directory "FIGURES" in the repositry root directory

 Copyright (C) 2021 Barbara Spitzer - spitzerb@mskcc.org
 This program is free software: you can redistribute it and/or modify 
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation, either version 3 of the License, or 
  any later version.

 This program is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU General Public License for more details.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r getLibs}
library(tidyverse)
library(ggforce)
library(grid)
library(gridExtra)
library(UpSetR)
library(lubridate)
library(formattable)
library(kableExtra)
library(tidytext)
library(reshape)
library(reshape2)
library(ggrepel)
library(ggpubr)
library(cowplot)
library(data.table)
library(RColorBrewer)
library(seqinr)
```

```{r getDF}
DF_FIGS=readRDS("../DATA/DF_FIGS.rds") %>% filter(RM_VAR=="no") %>% mutate(VariantKey=na_if(VariantKey, "NA_NA_NA_NA"))
#data for final/supp figures: only included patients, onco/likely vars

ALL_RECORDS=readRDS("../DATA/173_frozen.rds") %>% filter(RM_VAR=="no")
#all patients & all variants (including unknowns,but not removed). Use this for reported stats and calculations

trt.table=read.csv("../DATA/DATA_FILES/therapy_timetable_CLEAN.csv")
#treatment information for all patients

pt.cats  <- read.table("../DATA/DATA_FILES/pt_num_PID_cat.txt", header=TRUE) %>% mutate(pt_cat=case_when(pt_cat=="other" ~"transient", TRUE ~ as.character(pt_cat)))
pt.cats$PID.short <- substr(pt.cats$PID,5,10)
rownames(pt.cats) <- pt.cats$PID.short

facets <- readRDS("../DATA/FACETS_TIMELINES.rds") %>% filter(patient %in% pt.cats$PID.short)
facets$pt.cat <- pt.cats[facets$patient,"pt_cat"]

```

```{r 1b_oncobarplot}
#a barplot summary of oncoprint (fig 1a) found in basic_oncoprint.Rmd

theme_set(theme_bw(base_size=8))
pttypes <- ALL_RECORDS %>% group_by(patient.category.2) %>% summarize(npts=n_distinct(PID))
ncontrol <- pttypes %>% filter(patient.category.2=="control") %>% select(npts) %>% .$npts #21
ntransient <- pttypes %>% filter(patient.category.2=="transient") %>% select(npts) %>% .$npts #14
ntransform <- pttypes %>% filter(patient.category.2=="transformation") %>% select(npts) %>% .$npts #17

snvs <- DF_FIGS %>% filter(VAR_TYPE %in% c("sub", "indel"), sample.timepoint.2!="premalignant") %>% 
          group_by(patient.category.2) %>% summarize(snvs=n_distinct(PID))
snvcontrol <- snvs %>% filter(patient.category.2=="control") %>% select(snvs) %>% .$snvs #1
snvtransient <- snvs %>% filter(patient.category.2=="transient") %>% select(snvs) %>% .$snvs #4 
snvtransform <- snvs %>% filter(patient.category.2=="transformation") %>% select(snvs) %>% .$snvs #15

fus <- DF_FIGS %>% filter(VAR_TYPE=="fusion", sample.timepoint.2!="premalignant") %>% 
          group_by(patient.category.2) %>% summarize(fus=n_distinct(PID))
fuscontrol <- 0  #by definition
fustransient <- fus %>% filter(patient.category.2=="transient") %>% select(fus) %>% .$fus #5 
fustransform <- fus %>% filter(patient.category.2=="transformation") %>% select(fus) %>% .$fus #10

#need both FACETS and clinical CG results
#remove premalignant samples from FACETS results (reported only those in transformation smaples)
premalig <- ALL_RECORDS %>% filter(sample.timepoint.2=="premalignant") %>% select(TARGET_NAME) %>% distinct()

#facets
cnvs <- facets %>% filter(!sample %in% unlist(premalig)) #%>% group_by(pt.cat) %>% summarize(cnvs=n_distinct(patient))
#clinical cg
cnas <- read.csv("../DATA/DATA_FILES/p173_variants_SVs_for_timeplots_CLEAN.txt", check.names=FALSE, sep="\t",stringsAsFactors=FALSE, header=TRUE) %>%
        mutate(pt_type=case_when(pt_type=="other" ~"transient", TRUE ~ as.character(pt_type))) %>%
        filter(Algorithm=="CG_PUB", variant_type=="aneuploidy", CLONAL==TRUE, PID %in% pt.cats$PID, sample_type!="premalignant") 

cnvscontrol =0  #by definition
cnvstransient <- length(union(cnvs %>% filter(pt.cat=="transient") %>% distinct(patient) %>% unlist(),
                                cnas %>% filter(pt_type=="transient") %>% distinct(PID) %>% unlist() %>% substr(.,5,10)))
cnvstransform <- length(union(cnvs %>% filter(pt.cat=="transformation") %>% distinct(patient) %>% unlist(),
                                cnas %>% filter(pt_type=="transformation") %>% distinct(PID) %>% unlist() %>% substr(.,5,10)))


cnas <- read.csv("../DATA/DATA_FILES/p173_variants_SVs_for_timeplots_CLEAN.txt", check.names=FALSE, sep="\t",stringsAsFactors=FALSE, header=TRUE) %>%
        mutate(pt_type=case_when(pt_type=="other" ~"transient", TRUE ~ as.character(pt_type))) %>%
        filter(Algorithm=="CG_PUB", variant_type=="aneuploidy", CLONAL==TRUE, PID %in% pt.cats$PID) %>% 
        group_by(pt_type) %>% summarize(cnas=n_distinct(PID))

bardf=data.frame(proportion=c(snvtransform/ntransform,fustransform/ntransform,cnvstransform/ntransform,
                              snvtransient/ntransient,fustransient/ntransient,cnvstransient/ntransient,
                              snvcontrol/ncontrol,fuscontrol/ncontrol,cnvscontrol/ncontrol),
    type=c("snv","sv","cnv","snv","sv","cnv","snv","sv","cnv"),
    group=c("transformation","transformation","transformation","transient","transient","transient","control","control","control"))
#bardf=data.frame(proportion=c(15/17,10/17,10/17,4/14,5/14,10/14,1/21,0,0),type=c("snv","sv","cnv","snv","sv","cnv","snv","sv","cnv"),group=c("transformation","transformation","transformation","transient","transient","transient","control","control","control"))
bardf$group=factor(bardf$group,levels=c("transformation","transient","control"))
bardf$type=factor(bardf$type,levels=c("cnv","sv","snv"))

ggplot(bardf,aes(x=type,y=proportion,fill=type)) + geom_bar(stat='identity') + facet_wrap(~group, nrow=3) + coord_flip() + 
scale_fill_manual(values=c("lightsteelblue","darkorange","lightcyan3")) + scale_y_continuous(limits=c(0,1)) + scale_x_discrete(labels=c("CNVs", "SVs", "muts")) + theme(legend.position = 'none', strip.text=element_text(size=7, margin=margin(0.05,0,0.05,0,"cm")),legend.margin=margin(0,0,0,0), axis.text.x=element_text(size=7), axis.text.y=element_text(size=7), axis.title.x=element_text(size=8), axis.title.y=element_blank(), panel.spacing.y=unit(0.15,"cm"))

ggsave("fig1b_oncobar.pdf",device="pdf", path="../FIGURES/", width=4.5, height=2.5, useDingbats=FALSE)
```

```{r 1c_ddpcr}
theme_set(theme_minimal())
ddpcr_full=ddpcr_full %>% filter(!Assay=="TP53_C135S" & !sample. %in% c("7324","7291"))

d1=ddpcr_full %>% mutate(status=ifelse(Droplet.Count.Mu>1,"nonzero","zero")) %>%filter(leukgen_PID=="107649")
d2=ddpcr_full %>% mutate(status=ifelse(Droplet.Count.Mu>1,"nonzero","zero")) %>%filter(leukgen_PID=="118727")#,Assay=="TP53_V172F")
d3=ddpcr_full %>% mutate(status=ifelse(Droplet.Count.Mu>1,"nonzero","zero")) %>%filter(leukgen_PID=="118730")#,Assay=="TP53_R249S.2?")
d4=ddpcr_full %>% mutate(status=ifelse(Droplet.Count.Mu>1,"nonzero","zero")) %>%filter(leukgen_PID=="118731")#,Assay=="TP53_R213P")
d5=ddpcr_full %>% mutate(status=ifelse(Droplet.Count.Mu>1,"nonzero","zero")) %>%filter(leukgen_PID=="118732")#,Assay=="TP53_A276P")

nd1= length(unique(d1$Assay))
nd2= length(unique(d2$Assay))
nd3= length(unique(d3$Assay))
nd4= length(unique(d4$Assay))
nd5= length(unique(d5$Assay))

patlist=c(31,4,8,9,10)
ttaml <- trt.table %>% filter(patient %in% patlist, !is.na(TT_AML)) %>% select(TT_AML) %>% unlist()
ttaml <- ttaml[c(5,1,2,3,4)] #default order is numerical for patients (4,8,9...) - reorder as per current plot

#values manually copied from trt.table
tx <-   as.data.frame(rbind(c("d1",9,151,"chemo",nd1), rbind(c("d1",168,179,"rt",nd1),
        rbind(c("d2",4,74,"chemo",nd2), rbind(c("d2",350,358,"rt",nd2), rbind(c("d2",918,930,"rt",nd2), rbind(c("d2",931,942,"rt",nd2),
        rbind(c("d3",8,141,"chemo",nd3), rbind(c("d3",169,183,"chemo",nd3), rbind(c("d3",407,427, "rt",nd3),
        rbind(c("d4",6,144,"chemo",nd4), rbind(c("d4",187,194,"chemo",nd4), rbind(c("d4",1326,1420,"chemo",nd4), rbind(c("d4",1618,2549,"chemo",nd4), 
        rbind(c("d4",2854,2901,"chemo",nd4), rbind(c("d4",513,533,"rt",nd4), rbind(c("d4",1856,1876,"rt",nd4), rbind(c("d4",2042,2062,"rt",nd4),
        rbind(c("d5",52,159,"chemo",nd5), rbind(c("d5",1185,1199,"rt",nd5), rbind(c("d5",1136,1520,"chemo",nd5))))))))))))))))))))),stringsAsFactors=FALSE)
colnames(tx) <- c("pt","start","end","tx","n.assays")
tx$pt <- factor(tx$pt)
tx$tx <- factor(tx$tx, levels=c("chemo","rt"))
tx$start <- as.numeric(tx$start)
tx$end <- as.numeric(tx$end)
tx$n.assays <- as.numeric(tx$n.assays)

plotlist=list()
pts <- as.character(unique(tx$pt))
for(p in pts){
  d <- get(p)
  if(d$Assay[1]=="TP53_R249S.2?") d$Assay <- "TP53_R249S"
  txs <- tx %>% filter(pt==p)

 #add time of transformation to plot ; create separate data df for ttaml
  assays <- unique(d$Assay)
  d$ttaml <- ttaml[as.numeric(substr(p,2,2))]/30.42
  d$ttstatus <- "transformation"

  plot=ggplot() + geom_point(data=d, aes(x=months,y=Assay,color=log(Ratio...Mu...WT..+0.0001),shape=status),size=4) + 
  geom_point(data=d, aes(x=ttaml,y=Assay, shape=ttstatus), color="gray30", size=2) +  #add time of transformation to plot
  scale_x_continuous(limits=c(0,100)) + scale_color_gradient2(low="gray90",mid="lightcyan3",high="red",midpoint = -4,breaks=c(-6.9,-4.6,-2.3,-0.69),labels=c("0.001","0.01","0.1","0.5"))+ 
    scale_shape_manual(name="Status", values=c(nonzero=16,zero=1,transformation=3), labels=c("zero", "nonzero", "MDS/AL"),
                        guide=guide_legend(override.aes=list(shape=c(1,16,3),color=rep("gray30",3),size=c(4,4,2.5)))) +
    labs(title=d$leukgen_PID[1], y="",x="",color="VAF") + #labs(y="variant",color="log(VAF)") +
    geom_rect(data=txs, aes(xmin=start/30.42,xmax=end/30.42,ymin=n.assays+n.assays*0.3, ymax=n.assays+n.assays*0.6,fill=tx)) +
    scale_fill_manual(values=c(chemo=col2alpha("royalblue",0.6), rt=col2alpha("purple4",0.6))) + 
   # guides(shape=guide_legend(order=1), color=guide_colorbar(order=2), color=guide_legend(order=3)) +
    theme(axis.text.y=element_text(size=7), #plot.margin=unit(c(-0.2,0,-0.2,0),"cm"), 
      legend.text=element_text(size=7), legend.title=element_text(size=8),legend.key.size=unit(0.5,"cm"),
      plot.title=element_text(size=8, hjust=0.5, vjust=0))
  if(p=="d5") {plot=plot+theme(axis.text.x=element_text(size=7))} else {plot=plot+theme(axis.text.x=element_blank())}
  if(p=="d1") {plot=plot+theme(plot.margin=unit(c(0.2,0,-0.2,0),"cm"))} else {plot=plot+theme(plot.margin=unit(c(-0.2,0,-0.2,0),"cm"))}
  plotlist[[p]]=plot
}

gga <- ggarrange(plotlist[["d1"]],plotlist[["d2"]],plotlist[["d3"]],plotlist[["d4"]],plotlist[["d5"]],ncol=1,common.legend=TRUE, legend="bottom")
annotate_figure(gga, left=text_grob("variant", rot=90, size=8, vjust=2.2), bottom=text_grob("months", size=8,vjust=-0.9))

ggsave("fig1c_ddpcr_revision.pdf", device="pdf", path="../FIGURES/",width=5.6,height=4.6, useDingbats=FALSE)

```

```{r 2a_heatmap_gm}
vardf=DF_FIGS %>% filter(VAR_TYPE %in% c("sub","indel"),TARGET_VAF>0) %>% group_by(PID, sample_number) %>% 
  summarise(vars=length(unique(VARIANT)), vaf=max((TARGET_VAF)),type=unique(patient.category.2)) %>% 
  mutate(ps=paste0(PID,sample_number))
colnames(vardf)=c("PID", "samplenum","count","vaf","type","ps")

allsamples= ALL_RECORDS %>% filter(REMOVE=="no") %>% mutate(ps=paste0(PID,sample_number)) %>% 
  select(TARGET_NAME,ps,sample_number,patient.category.2) %>% unique()

newtab=merge(allsamples,vardf, by.x=c('ps','sample_number','patient.category.2'), by.y=c('ps','samplenum','type'), 
  all.x=TRUE) %>% mutate(final=ifelse(is.na(PID),0,count),patient=substr(ps,1,10))
newtab$sample_number=as.integer(newtab$sample_number)
nt=newtab %>% group_by(patient) %>% mutate(nsamp=length(patient)) #to change patient order by dec sample count
nt=nt[order(nt$nsamp),]

###use this factoring to order by sample count
nt$patient=factor(nt$patient,levels=c(unique(nt$patient)))
hmorderlist=nt$patient

theme_set(theme_minimal()) #fill color by max vaf
nt$patient.category.2=factor(nt$patient.category.2,levels=c("transformation","transient","control"))
C=nt %>% filter(patient.category.2=="transformation") %>% ggplot(aes(x=sample_number, y=patient, fill=log(vaf))) + 
  geom_tile(color="black") + facet_wrap(~patient.category.2, scales="free_y")  +
  labs(title="Mutational timeline: GMs", x="sample number" , y="patient ID", fill="log(max VAF)") + 
  theme(plot.title=element_text(hjust = 0.5, size=8, margin=margin(0,0,3,0)),legend.position = "bottom", legend.title = element_text(size=7, vjust=0.75), legend.text=element_text(size=7), legend.key.height = unit(0.5,"cm"), axis.text=element_text(size=7), axis.title=element_text(size=7), legend.margin=margin(0,0,0,0), strip.text = element_blank()) + 
  scale_fill_gradient2(low="gray95",mid="lightsteelblue",high="darkorange",na.value = "white",midpoint=-2.6) + scale_x_continuous(breaks=c(2,4,6,8))

ggsave("fig2a_heatmapGM.pdf",device="pdf", path="../FIGURES/",width=2.8,height=4.5, useDingbats=FALSE)
```

```{r 2b_heatmap_fus}
# missing * legend = KMT2A fusion
vardf=DF_FIGS %>% filter(VAR_TYPE=="fusion",TARGET_VAF>=0.005) %>% rowwise() %>% mutate(fus_type=ifelse(any(str_detect(VARIANT,c("KMT2A","11q23","4,11","4;11"))),1,0)) %>% 
  group_by(PID, sample_number) %>% summarise(length(unique(VARIANT)), max(as.numeric(TARGET_VAF)),unique(patient.category.2),sum(fus_type)) %>% 
  mutate(ps=paste0(PID,sample_number))
colnames(vardf)=c("PID", "samplenum","count","vaf","type","fus_type","ps")

newtab=merge(allsamples,vardf, by.x='ps', by.y='ps', all.x=TRUE)
nt=newtab %>% mutate(final=ifelse(is.na(PID),0,count),patient=substr(ps,1,10),fus_type=ifelse(is.na(fus_type),0,fus_type)) %>% mutate(mll=ifelse(fus_type>0,"MLL","other"))

nt=nt %>% group_by(patient) %>% mutate(nsamp=length(patient))
nt=nt[order(nt$nsamp),]
nt$patient=factor(nt$patient,levels=c(unique(nt$patient)))
nt$sample_number=as.integer(nt$sample_number)

B=nt %>% filter(patient.category.2=="transformation") %>%
  ggplot(aes(x=sample_number, y=patient, fill=count/count)) + 
    geom_tile(color="black",alpha=1) + facet_wrap(~patient.category.2, scales="free_y") + geom_point(aes(size=mll),color="gray50") +
    labs(title="Mutational timeline: fusions", x="sample number", y="patient ID", fill="fusion detected") + 
    theme(plot.title=element_text(size=8, hjust = 0.5, margin=margin(0,0,3,0)),legend.position = "bottom", legend.title = element_text(size=7, vjust=0.75), legend.key.height = unit(0.5,"cm"), legend.key.width=unit(0.5,"cm"), axis.text=element_text(size= 7), axis.title=element_text(size=7), legend.margin=margin(0,0,0,0), strip.text = element_blank()) + 
    scale_fill_gradient(low="lightcyan3",high="lightcyan3", na.value="white", label="") + scale_x_continuous(breaks=c(2,4,6,8)) + scale_size_manual(values=c(1, NA),guide="none")

# missing * legend = KMT2A fusion (added manually)
ggsave("fig2b_heatmapFusions.pdf", device="pdf", path="../FIGURES/", width=2.8, height=4.5, useDingbats=FALSE)

```

```{r 2c_swimmers}
var_trans <- DF_FIGS %>% filter(TMN_VAR=="yes", !grepl("t\\(", GENE), !grepl("der\\(", GENE), !grepl("11q23", GENE)) %>% 
                distinct(VARIANT) %>% unlist()
vars_at_tmn=DF_FIGS %>% filter(VARIANT %in% var_trans,TARGET_VAF>=0.02) #all variants that are also present at tmn

summed=vars_at_tmn %>% group_by(PID,sample.time.before.transf) %>% summarise(count=length(unique(VARIANT)),mvaf=max(TARGET_VAF)) #aggr by timepoint

mint=summed %>% group_by(PID) %>% filter(abs(sample.time.before.transf)== min(abs(sample.time.before.transf))) %>% mutate(sample.time.before.transf=0) #closest to tmn
maxt=summed %>% group_by(PID) %>% filter(sample.time.before.transf== max(sample.time.before.transf)) #earliest detec
othert=summed %>% group_by(PID) %>% filter(sample.time.before.transf<max(sample.time.before.transf) & sample.time.before.transf>0) %>% mutate(count=100)#between min and max
##modify the count var of othert to allow for bold where muts gained

priors=data.frame()
##get samples prior to mutation detection for pts
for(person in unique(vars_at_tmn$PID)){
  mt=maxt[maxt$PID==person,'sample.time.before.transf']
  samples=ALL_RECORDS %>% filter(PID==person, sample.time.before.transf>mt$sample.time.before.transf) %>% 
            select(PID,sample.time.before.transf) %>% mutate(count=0,mvaf=0)
  priors=rbind(priors,samples)
}

wtfall=as.data.frame(rbind(mint,maxt,othert))
x=rbind(wtfall,priors)
#x$PID=factor(x$PID,levels=levels(porder))
x=x %>% mutate(found=ifelse(count==0,"neg",ifelse(count==100,"pos","first")))

#last 199
#yes/no denotes acquisition of new mutation in a 2nd+ positive sample
x$found2=c(rep("no",34),"yes","no","yes","no","no","no","no","yes","no","no","no","yes","no","no",rep("no",182))
x=x %>% rowwise() %>% mutate(found2=ifelse(found=="pos"&found2=="yes","+ hit",ifelse(found=="first","pos",found)))
x$found2 <- factor(x$found2, levels=c("neg", "pos", "+ hit"))
#order patients based on time at first detection (earliest to latest)
ptorder <- x %>% filter(found!="neg") %>% group_by(PID) %>% summarize(maxt=max(sample.time.before.transf)) %>% 
              arrange(maxt) %>% ungroup() %>% distinct(PID) %>% unlist()
x$PID <- factor(x$PID, levels=ptorder)


theme_set(theme_minimal())
x %>% unique() %>% filter(sample.time.before.transf<65) %>% unique() -> x.plot
x.plot %>% ggplot(aes(x=-sample.time.before.transf,y=PID,color=mvaf,shape=found2)) + geom_point(size=3) + 
    scale_color_gradient2(low="gray90",mid="lightsteelblue",high="darkorange") + labs(x="months to tMN",shape="detection",color="max vaf") + 
    geom_vline(xintercept=0,lwd=0.3,color="gray35") + theme(legend.position = 'bottom') + scale_shape_manual(values=c(1,19,17)) + 
    annotate("segment", x = -61.8, xend = 0, y = 17, yend = 17, colour = "gray30",size=0.2) + 
    annotate("segment", x = -30.1, xend = 0, y = 16, yend = 16,colour = "gray30",size=0.2) + 
    annotate("segment", x = -28.5, xend = 0, y = 15, yend = 15,colour = "gray30",size=0.2) + 
    annotate("segment", x = -26.2, xend = 0, y = 14, yend = 14,colour = "gray30",size=0.2) + 
    annotate("segment", x = -24.7, xend = 0, y = 13, yend = 13,colour = "gray30",size=0.2) + 
    annotate("segment", x = -16.3, xend = 0, y = 12, yend = 12,colour = "gray30",size=0.2)  + 
    annotate("segment", x = -15.7, xend = 0, y = 11, yend = 11,colour = "gray30",size=0.2) + 
    annotate("segment", x = -11.8, xend = 0, y = 10, yend = 10,colour = "gray30",size=0.2) + 
    annotate("segment", x = -8.7, xend = 0, y = 9, yend = 9,colour = "gray30",size=0.2)  + 
    annotate("segment", x = -8.7, xend = 0, y = 8, yend = 8,colour = "gray30",size=0.2)  + 
    annotate("segment", x = -3.0, xend = 0, y = 7, yend = 7,colour = "gray30", size=0.2) + 
    annotate("segment", x = -2.3, xend = 0, y = 6, yend = 6,colour = "gray30", size=0.2) + 
    annotate("segment", x = -1.3, xend = 0, y = 5, yend = 5,colour = "gray30", size=0.2) + 
    geom_text_repel(data = subset(x, found=="first" & sample.time.before.transf>=0),aes(label = count),color="black",size=3,nudge_y = 0.3,segment.color = NA,nudge_x = 0.5) + 
    theme(axis.text=element_text(size=7),axis.title=element_text(size=8),legend.text = element_text(size=7),legend.title=element_text(size=8,vjust=0.75),legend.key.size=unit(0.5, "cm"), legend.margin=margin(1,1,1,1))

ggsave("fig2c_swimmers.pdf",device="pdf", path="../FIGURES/",width=5.5,height=5, useDingbats=FALSE)
```

```{r 4a_cnbarplot}
trans_list=c("I-H-107650","I-H-107651","I-H-133427","I-H-133429","I-H-133430","I-H-133431","I-H-133432","I-H-133433","I-H-133434","I-H-133435","I-H-133436","I-H-133437","I-H-133522","I-H-133523")

tmn.samps <- unique(ALL_RECORDS[ALL_RECORDS$sample.timepoint.2=="tMN","TARGET_NAME"])
trans.samps <- unique(ALL_RECORDS[ALL_RECORDS$sample.timepoint.2=="transient","TARGET_NAME"])

cnin=readRDS("../DATA/FACETS_TIMELINES.rds") %>% mutate(chr_arm=paste(chrom,arm,sep="")) %>% unique()
cnvs=cnin %>% select(isablID=sample,chr_arm,type) %>% mutate(id=chr_arm) %>% filter(isablID %in% c(tmn.samps,trans.samps))

colnames(cnvs)[colnames(cnvs) == 'isablID'] <- 'TARGET_NAME'
colnames(cnvs)[colnames(cnvs) == 'id'] <- 'VAR_ID'

cnvs= cnvs %>% mutate(EFFECT=ifelse(cnvs$type=="amp","AMP",ifelse(cnvs$type=="del","DEL","LOH")),TARGET_NAME=substr(TARGET_NAME,1,10)) %>% 
                select(TARGET_NAME,VAR_ID,EFFECT)

CG=read.table("../DATA/DATA_FILES/p173_variants_SVs_for_timeplots_CLEAN.txt",header=T,sep="\t")
CG$CG_EFFECT <- sapply(CG$CG_REDUCED, FUN=function(x) { e=substr(x,nchar(as.character(x)),nchar(as.character(x)))
                                                        ifelse(e=="-", "DEL",ifelse(e=="+","GAIN","")) 
                                                      } )
CG$arm <- sapply(CG$CG_REDUCED, FUN=function(x) { a=substr(x,nchar(as.character(x))-1,nchar(as.character(x))-1) 
                                                  ifelse(a %in% c("p","q"),substr(x,1,nchar(as.character(x))-1),"")
                                                } )
CG$TARGET_NAME=as.character(CG$TARGET_NAME)
CG$variant=as.character(CG$variant)

# not including MLL as it is not aneuploidy of a whole arm
CG_cna=CG%>% filter(Algorithm=="CG_PUB", variant_type=="aneuploidy", CLONAL==TRUE, !(arm %in% c("","11q23")), 
                    (pt_type %in% c("transient", "other") | TARGET_NAME %in% tmn.samps)) %>%  #either tMN sample for transformation pts or any transient pt sample
              rowwise() %>% 
  mutate(TARGET_NAME=substr(TARGET_NAME,1,10),CG_EFFECT=ifelse(CG_EFFECT=="GAIN","AMP",CG_EFFECT)) %>% 
  select(TARGET_NAME,CG_ARM=arm,CG_EFFECT) %>% unique()

colnames(CG_cna)=c("TARGET_NAME","VAR_ID","EFFECT")

rm_list=c("I-H-118739","I-H-118740", "I-H-107652")

cnvs_all=rbind(cnvs,CG_cna) %>% filter(VAR_ID!="", !(TARGET_NAME %in% rm_list)) %>% unique()

theme_set(theme_minimal())
cnvs_all_groups = cnvs_all %>% rowwise %>% mutate(group=ifelse(TARGET_NAME %in% trans_list,"transient",ifelse(TARGET_NAME %in% rm_list, "remove","transformation"))) %>% 
  unique() %>% filter(group!="remove") %>% group_by(VAR_ID,EFFECT,group) %>% summarise(n=length(EFFECT)) %>% 
  ungroup() %>% rowwise() %>% mutate(n=ifelse(group=="transformation",n*100/17,n*100/14))

cnvs_all_groups$VAR_ID=factor(cnvs_all_groups$VAR_ID,levels=rev(c("1p","1q","2p","2q","3p","3q","4p","4q","5p","5q","6p","6q","7p","7q","8p","8q","9p","9q","10p","10q","11p","11q","12p","12q","13p","13q","14p","14q","15p","15q","16p","16q","17p","17q","18p","18q","19p","19q","20p","20q","21p","21q","22p","22q","Xp","Xq")))
cnvs_plot=cnvs_all_groups %>% complete(group,VAR_ID,EFFECT) %>% rowwise() %>% mutate(n=ifelse(is.na(n),0,n)) %>% filter(!is.na(VAR_ID))
cnvs_plot %>% ggplot(aes(x=VAR_ID)) + geom_bar(data=subset(cnvs_plot,group=="transient"),aes(x=VAR_ID,y=n,fill=EFFECT),stat='identity',color="white") + 
geom_bar(data=subset(cnvs_plot,group=="transformation"),aes(x=VAR_ID,y=-n,fill=EFFECT),stat='identity',color="white") + 
  scale_y_continuous(limits=c(-40,40),labels=c(40,20,0,20,40)) + theme(legend.position = 'bottom',plot.title=element_text(size=8,hjust=0.5), 
    axis.text=element_text(size=7), axis.title=element_text(size=8), legend.title=element_text(size=8,vjust=0.8), legend.text=element_text(size=7), 
    legend.key.size=unit(c(0.5,0.5),"cm"), legend.margin=margin(0,0,0,0)) +
  labs(x="locus",y="patient proportion (%) ",title="transformation                                       transient") + 
  scale_fill_manual(values=c("lightsteelblue","lightsalmon","paleturquoise3","red")) + geom_hline(yintercept = 0,color="gray50",lwd=0.7) + coord_flip()

ggsave("fig4a_cnbarplot_updated.pdf",device="pdf", path="../FIGURES/", width=4.2,height=7, useDingbats=FALSE)
```

```{r 4bc_therapymuts, fig.width=4, fig.height=7}
#SUMMARY OF THERAPY RECEIVED AND MUTATION BURDER BY COHORT

#mutational burden
burden=DF_FIGS %>% filter(TARGET_VAF>=0.02) %>% group_by(TARGET_NAME) %>% summarise(muts=length(unique(VARIANT)),time=unique(sample.time..mo.), type=unique(patient.category.2)) %>% ungroup

mutated_samps=unique(burden$TARGET_NAME)
tobind=ALL_RECORDS[(which(!ALL_RECORDS$TARGET_NAME %in% mutated_samps)),] %>% mutate(muts=0) %>% 
  select(TARGET_NAME,muts,time=sample.time..mo.,type=patient.category.2)
burden=rbind(burden,tobind) %>% unique()

theme_set(theme_minimal()) #boxplot of mutation counts by sample by group

mut_burd_plot=ggplot(burden, aes(x=type, y=muts, color=type)) +
  geom_boxplot(alpha=0.65, size=0.8) + geom_jitter(data=subset(burden,muts==0),alpha=0.25,height=0.25) + 
    geom_jitter(data=subset(burden,muts>0),alpha=0.8,height=0.25) + 
    labs(x="patient category", y="mutation count",title="mutation count by sample",size="years post Dx") +
    theme(axis.text=element_text(size=7), axis.ticks.x = element_blank(), legend.position="none", plot.title=element_text(hjust=0.5, size=8),
      axis.title=element_text(size=8)) +
  scale_color_manual(values=c("salmon","thistle4","olivedrab3")) + scale_y_continuous(breaks=c(2,4,6,8,10)) + 
  stat_compare_means(method="t.test",label.y = c(7,7),comparisons = list(c("control","transformation"),c("transformation","transient")),bracket.size = 0.1, size=2.8) + 
  scale_size(range=c(0.6,3.8),breaks=c(1,5,10,15))

#calculate amount of therapy received in each patient group
trt.table=read.csv("../DATA/DATA_FILES/therapy_timetable_CLEAN.csv")
trt_sums=c() #list of treatment durations, for all patients

#1533 days threshold
sum_therapy=function(trtfile){ #function to sum therapy given details subsetted for 1 patient
  duration=0
    for(i in 1:nrow(trtfile)) {
        for(therapy in c("chemo","RT")){
        if(!is.na(trtfile[i, paste0(therapy,".TTS")]) & trtfile[i,paste0(therapy,".TTS")]<1533){
          x0=trtfile[i,paste0(therapy,".TTS")]
          x1=trtfile[i,paste0(therapy,".TTE")]
          duration=duration+(x1-x0)
        }
        else{
          duration=duration
          }
        }
    }
  return(duration)
}

for(pt in c(1:71)){
  trtfile=trt.table %>% filter(patient==pt)
  trt_sums=c(trt_sums,(sum_therapy(trtfile)/30)) #number of months of therapy for that patient
}

therapy_df=data.frame(patient=unique(trt.table$patient),therapy_duration=trt_sums)
clindata=ALL_RECORDS %>% select(patient.category.2,PID,sample.time.before.transf,Patient...x) %>% 
  group_by(PID) %>% summarise(pat=unique(Patient...x),ptcat=unique(patient.category.2),ptid=unique(PID),transtime=max(sample.time.before.transf))

trt_df=cbind(therapy_df %>% filter(patient %in% clindata$pat),clindata)

trt_df %>% filter(!patient %in% c(2,6,14,15,16,17,22,28,30,34,36,37,38,43,46,50,53,64,65)) -> trt_df_plot

treat_burd_plot = ggplot(trt_df_plot, aes(x=ptcat,y=therapy_duration, fill=ptcat,color=ptcat)) + 
  geom_boxplot(alpha=0.65,size=0.8) + scale_fill_manual(values=c("salmon","thistle4","olivedrab3")) + 
  scale_color_manual(values=c("salmon","thistle4","olivedrab3")) + 
  labs(x="patient category",y="therapy duration (months RT/chemo)",title="cumulative therapy duration") + 
  theme(plot.title = element_text(hjust=0.5, size=8), legend.position = "none", axis.title=element_text(size=8), axis.text=element_text(size=7)) + 
  stat_compare_means(method='t.test',comparisons = list(c("transformation","control"),c("transformation","transient")),label.y=c(50,50),bracket.size=0.1, size=2.8)

grid.arrange(treat_burd_plot,mut_burd_plot,nrow=2)  # ncol or nrow as appropriate for setup of figure
a <- arrangeGrob(treat_burd_plot, mut_burd_plot,nrow=2)

ggsave("fig4bc_therapy_mutburden.pdf", a, device="pdf", path="../FIGURES/", width=4, height=7, useDingbats=FALSE)
```

```{r 5a_IMPACT_tumortypes}
#impact plots
ipeds.plot <- readRDS("../DATA/IMPACT_PLOTS.rds")

ipeds.plot$Dx.Category <- factor(ipeds.plot$Dx.Category, levels=names(sort(table(ipeds.plot$Dx.Category),decreasing=TRUE)))

ggplot(ipeds.plot, aes(x=Dx.Category)) + 
  geom_bar(stat="count", width=0.75, fill="cornflowerblue") + theme_minimal(base_family='Helvetica') + 
  theme(plot.title = element_text(hjust = 0.5, face='bold'), title=element_text(size=8), legend.title=element_text(size=8), legend.text=element_text(size=7), 
    strip.text = element_text(size=8), axis.text.x=element_text(size=7, angle=45,hjust=1), axis.text.y=element_text(size=7), axis.title.y=element_text(size=8), 
    axis.title.x=element_blank(), legend.key.height=unit(0.75,"cm")) + 
    scale_y_continuous(expand=c(0,0)) + ggtitle("IMPACT cohort based on tumor type")

ggsave("fig5a_impact_tumortype.pdf", path="../FIGURES/", device="pdf", width=4, height=4, useDingbats=FALSE)
```

```{r 5b_IMPACT_ages}
ggplot(ipeds.plot, aes(x=age_grp)) + 
  geom_bar(stat="count", width=0.75, fill="lightsteelblue") + theme_minimal(base_family='Helvetica') + 
  theme(plot.title = element_text(hjust = 0.5, face='bold'), title=element_text(size=8), legend.title=element_text(size=8), legend.text=element_text(size=7), 
    strip.text = element_text(size=8), axis.text.x=element_text(size=7), axis.text.y=element_text(size=7), axis.title=element_text(size=8), legend.key.height=unit(0.75,"cm")) +
  scale_y_continuous(expand=c(0,0)) + ggtitle("IMPACT cohort based on age") + labs(x="Age Group") + scale_x_discrete(labels=c("0-5", ">5-10", ">10-15", ">15-20", ">20-25"))

ggsave("fig5b_impact_ages_barplot.pdf", path="../FIGURES/", device="pdf", width=3.8, height=3.8, useDingbats=FALSE)

```

```{r 5c_IMPACT_chgenes_pdonly}

#same as Fig S11
ch.peds <- readRDS("../DATA/IMPACT_VAFS.rds")

genes.pedsch <- substr(colnames(ch.peds)[3:ncol(ch.peds)],5,12)
genes.chpd <- substr(names(which(colSums(ch.peds[which(ch.peds$ch_my_pd>0),3:ncol(ch.peds)])>0)),5,12)
ch.peds[ch.peds==0] <- NA


ch.cols <- rep("grey20", length(genes.pedsch))
names(ch.cols) <- genes.pedsch
ch.cols[names(ch.cols) %in% genes.chpd] <- "#ce1256"

ch.peds.melt <- melt(data.table(ch.peds[,3:ncol(ch.peds)]))
###

chpd <- ch.peds.melt[which(substr(ch.peds.melt$variable,5,12) %in% genes.chpd),] 
ggplot(chpd, aes(x=variable, y=value)) + geom_boxplot(lwd=0.3) +  geom_jitter(width=0) +
  theme_minimal() +  scale_x_discrete(labels=genes.chpd, expand=c(0,0)) + 
    theme(axis.text.x=element_text(angle=45, hjust=1), axis.text=element_text(size=7), 
      axis.title=element_text(size=8), plot.title=element_text(hjust=0.5, size=8)) +
    xlab("Gene") + ylab("VAF") + ggtitle("CH involving oncogenic mutations in myeloid genes")

ggsave(filename="fig5c_impact_chPD_VAF.pdf", path="../FIGURES/", device="pdf", width=3.65, height=2.8, useDingbats=FALSE)

```

```{r 5d_CH-pd_piechart}

ipeds.plot$ch_final <- ifelse(ipeds.plot$ch_my_pd, "CH-PD", ifelse(ipeds.plot$CH_nonsilent, "CH non-PD", "No CH"))
ipeds.plot$ch_final <- factor(ipeds.plot$ch_final, levels=rev(c("CH-PD", "CH non-PD", "No CH")))

npeds <- nrow(ipeds.plot)
ipeds.tbl.ch <- table(ipeds.plot$ch_final)
tbl.prop <- ipeds.tbl.ch/npeds

ch.prop <- data.frame(t(rbind(ipeds.tbl.ch,tbl.prop)))
colnames(ch.prop) <- c("num", "prop")
ch.prop$categ <- rownames(ch.prop)

ch.prop <- ch.prop %>%
  arrange(desc(categ)) %>%
  mutate(lab.ypos = cumsum(prop) - 0.5*prop)
ggplot(ch.prop, aes(x = "", y = prop, fill = categ)) +
  geom_bar(width = 1, stat = "identity") +
  coord_polar("y", start = 0)+
  geom_text(aes(y = lab.ypos, label = round(prop,2)))+
  scale_fill_manual(values = c("powderblue", "sienna2","#6b517b")) +
  theme_void()

ggsave("fig5d_impact_piechart_RAW.pdf", path="../FIGURES/", device="pdf", useDingbats=FALSE)
```

```{r 5e_CH-pd_byage}
# CH-pd by age: main
ipeds.tbl <- table(ipeds.plot[,c("age_grp", "ch_my_pd")])
ipeds.plot$chpd <- sapply(ipeds.plot$ch_my_pd, FUN=function(x) {if(x) TRUE else FALSE } )

ipeds.tbl <- table(ipeds.plot[,c("age_grp", "ch_my_pd", "CH_nonsilent")])

ggplot(ipeds.plot, aes(fill=factor(ch_final), x=age_grp)) +
  geom_bar(position="fill", width=0.75) + 
  scale_fill_manual(values=c("#6b517b", "powderblue", "sienna2")) + 
  theme_minimal(base_family='Helvetica') + theme(axis.text=element_text(size=7), axis.title=element_text(size=8), legend.text=element_text(size=8), 
    legend.key.height=unit(0.5, "cm"), legend.position="left", legend.justification="bottom") +
  labs(fill=element_blank()) + xlab("Age Group") + scale_x_discrete(expand=c(0,0), labels=c("0-5", ">5-10", ">10-15", ">15-20", ">20-25")) + 
  ylab("% CH") + scale_y_continuous(expand=c(0,0))

ggsave("fig5e_impact_chpd_revision.pdf", path="../FIGURES/", device="pdf", height=2.8, width=4.3, useDingbats=FALSE)

```







