---
title: "sup_figures"
output: html_document
---

  
 Generates all supplemental figures for the manuscript 
 (except Fig S5, generated by therapy_timelines.R)
    Output should go in directory "FIGURES" in the repositry root directory

 Copyright (C) 2021 Barbara Spitzer - spitzerb@mskcc.org
 This program is free software: you can redistribute it and/or modify 
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation, either version 3 of the License, or 
  any later version.

 This program is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU General Public License for more details.


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r getLibs}
library(tidyverse)
library(ggforce)
library(grid)
library(gridExtra)
library(UpSetR)
library(lubridate)
library(formattable)
library(kableExtra)
library(tidytext)
library(reshape)
library(reshape2)
library(ggrepel)
library(ggpubr)
library(cowplot)
library(data.table)
library(RColorBrewer)
library(seqinr)
```

```{r getDF}
#data for final/supp figures: only included patients, onco/likely vars
DF_FIGS=readRDS("../DATA/DF_FIGS.rds") %>% filter(RM_VAR=="no")

#all patients (including removed, flagged) & all variants (including unknowns,but not removed). Use this for reported stats and calculations
ALL_RECORDS=readRDS("../DATA/173_frozen.rds") %>% filter(RM_VAR=="no")

#treatment information for all patients
trt.table=read.csv("../DATA/DATA_FILES/therapy_timetable_CLEAN.csv")
```

```{r s1a_samplesummary}
#SUMMARY OF SAMPLE TYPES - PREMALIGNANT, TMN, CONTROL

theme_set(theme_bw())
summary=ALL_RECORDS %>% filter(REMOVE=="no") %>% select(TARGET_NAME,PID,sample.timepoint.2,patient.category.2) %>% 
  unique() %>% select(PID,sample.timepoint.2,patient.category.2)

#modifies the x order to be by count, so bars are inc height
xord = summary %>% group_by(PID,sample.timepoint.2) %>% summarise(count=length(sample.timepoint.2)) 
summ=merge(xord,summary,by=c('PID','sample.timepoint.2')) %>% unique()
summ=summ[order(summ$sample.timepoint.2,summ$count),]
summ$PID=factor(summ$PID, levels=unique(summ$PID))
summ$patient.category.2=factor(summ$patient.category.2,levels=c("control","transient","transformation"))
summ$sample.timepoint.2=factor(summ$sample.timepoint.2,levels=c("control","transient","premalignant","tMN"))

ggplot(summ,aes(x=PID,y=count,fill=sample.timepoint.2,color=sample.timepoint.2)) + geom_bar(alpha=0.7, stat = 'identity') + facet_wrap(~patient.category.2, scales="free_x") + 
  theme(panel.grid.major.y=element_line(size=0.5),panel.grid.minor.y=element_line(size=0.5),panel.grid.major.x=element_blank(), legend.position="bottom", legend.text=element_text(size=7), 
    legend.title=element_text(size=8), plot.subtitle = element_text(size=7,hjust=0.5), plot.title=element_text(hjust=0.5, size=8),axis.title=element_text(size=8),axis.text.x = element_blank(), 
    axis.text.y=element_text(size=7), strip.text=element_text(size=7), axis.ticks.x = element_blank(), legend.key.size=unit(0.5, "cm"), legend.margin=margin(c(0,0,0,0))) + 
  labs(title="Sample summary", subtitle = "(samples = 199; patients = 52)" , x="patient", y="sample count", fill="sample category") + 
  scale_fill_manual(values=c("thistle4","ivory3","lightcyan3","lightsalmon"),labels=c("control","transient","premalignant", "t-MDS/AL")) + 
  scale_color_manual(values=c("thistle4","ivory3","lightcyan3","lightsalmon"),guide=FALSE) +
  scale_y_continuous(expand = c(0,0),breaks=c(2,4,6,8,10))

ggsave("figs1_samplesummary.pdf",device="pdf", path="../FIGURES/", width=6, height=4, useDingbats=FALSE)
```

```{r s1b_cgsummary}
#SUMMARY OF CLINICAL CGs

x=ALL_RECORDS %>% filter(REMOVE=="no") %>% select(TARGET_NAME,PID,patient.category.2,sample.CG) %>% unique()
y= x %>% select(TARGET_NAME,PID) %>% unique() %>% group_by(PID) %>% summarise(count=length(PID))
z=merge(x,y,by='PID') %>% arrange(count)
z$PID=factor(z$PID,levels=unique(z$PID))
z$patient.category.2=factor(z$patient.category.2,levels=c("control","transient","transformation"))
z$sample.CG=factor(z$sample.CG,levels=c("nl","abnl","unk"))

ggplot(z,aes(x=PID,fill=sample.CG)) + geom_bar(alpha=0.9,color="white") + labs(x="patient",fill="cytogenetics") + 
  facet_wrap(~patient.category.2, scales='free_x') + scale_fill_manual(values=c("lightcyan3","thistle4","ivory2"), labels=c("normal", "abnormal", "unknown"))+ 
   theme(axis.text.x = element_blank(), axis.text.y=element_text(size=7), axis.title=element_text(size=8),  
     axis.ticks.x=element_blank(), panel.grid.major.x = element_blank(), panel.grid.major.y=element_line(size=0.5), 
     panel.grid.minor.y=element_line(size=0.5), strip.text=element_text(size=7), legend.position="bottom", 
     legend.title=element_text(size=8), legend.text=element_text(size=7), legend.key.size=unit(0.5, "cm"), legend.margin=margin(c(0,0,0,0))) +
    scale_y_continuous(breaks=c(2,4,6,8,10),expand=c(0,0))
  
ggsave("figs2_cgsummary.pdf",device="pdf", path="../FIGURES/", width=6, height=4, useDingbats=FALSE)
```

```{r s1c_cvg}
#TARGETED SAMPLE COVERAGE
cov_file=read.csv("../DATA/DATA_FILES/p173_QC.csv") #coverage (original and topup together, labelled by batch)

mean_cov=data.frame(samples=cov_file$Sample.Name, cvg=cov_file$Mean.target.coverage,batch=substr(cov_file$Sample.Name,20,20),
  patient=substr(cov_file$Sample.Name,5,10)) %>% filter(batch==2,!patient %in% c('118739','133428','107652','118740')) #get relevant columns from coverage file, put in dataframe
val2=mean_cov %>% summarise(mean(cvg)) #mean cvg across samples post-topup

mean_cov %>% #only want batch 2 (post topup) coverage, pipe to ggplot
ggplot(aes(x=samples, y=cvg, fill=batch)) +
  geom_bar(stat="identity", alpha=0.6,color="lightcyan3") +
  scale_fill_manual(values=c("lightcyan3"), aesthetics = "fill")  +
  labs(title="Targeted Sample Coverage",x="sample",y="mean coverage") +
  theme(panel.grid = element_blank(), legend.position = "none", axis.text.x = element_blank(), axis.text.y=element_text(size=7),
    axis.ticks.x = element_blank(), axis.title=element_text(size=8), plot.title = element_text(hjust=0.5, size=8)) +
  scale_y_continuous(expand=c(0,0),limits=c(0,1350), breaks=c(0,250,500,717,1000,1250)) +
  geom_hline(yintercept=val2[[1]], linetype="dashed")

ggsave("figs3_coverage.pdf",device="pdf", path="../FIGURES/", width=6, height=4, useDingbats=FALSE)
```

```{r s2_genebar}
p=DF_FIGS %>% filter(VAR_TYPE %in% c("sub","indel")) %>% group_by(PID,VARIANT) %>% summarise(gene=as.character(unique(GENE)),cat=unique(patient.category.2)) %>% rowwise %>% mutate(gene=ifelse(str_detect(gene,"KMT2A"),"MLL fusion",gene))
# set order for displaying more commonly mutated genes
p$gene=factor(p$gene,levels=unique(c("TP53","MLL fusion","PPM1D","RUNX1","WT1","KRAS","PTPN11","BCOR",p$gene)))
p$cat <- factor(p$cat, levels=c("transformation", "transient", "control"))

p %>%
  ggplot(aes(x=gene,fill=cat))+ geom_bar(color="white",alpha=0.9) + 
  theme(axis.text.x = element_text(angle=90, size=7), axis.text.y=element_text(size=7), axis.title=element_text(size=8), 
    plot.title = element_text(hjust=0.5, size=8),legend.position='bottom', legend.margin=margin(c(0,0,0,0)), 
    legend.title=element_text(size=8), legend.text=element_text(size=7), legend.key.size=unit(0.5,"cm")) + 
  labs(x="gene",y="count (unique variants)",fill="group",title = "variant frequency by cohort") + 
  scale_fill_manual(values=c("thistle4","olivedrab3","lightsalmon")) + 
  scale_y_continuous(breaks = c(2,4,6,8,10))

ggsave("figs4_genebar.pdf",device="pdf", path="../FIGURES/", width=6, height=4, useDingbats=FALSE)
```

```{r ddpcr_readfile}
# read in ddpcr file
ddpcr_outfile=read.csv("../DATA/DATA_FILES/ddpcr_outfile.csv") %>% mutate(Droplet.Count.Mu=ifelse(Droplet.Count.Mu==1,0,Droplet.Count.Mu))
ddpcr_outfile$Ratio...Mu...WT..=as.numeric(as.character(ddpcr_outfile$Droplet.Count.Mu))/as.numeric(as.character(ddpcr_outfile$Droplet.Count.WT))

```

```{r ddpcr_preprocessing}
pvars=as.vector(unique(ddpcr_outfile$Assay)) #list of variants
colslist=c("slategray","lightcyan3","thistle4","darkorange","olivedrab3","cornflowerblue","lightpink","slateblue","olivedrab4") #some colors

ymaxs=c(0.001,0.001,0.006,0.01,0.003,0.01,0.003,0.025,0.008) #different y ranges to scale each zoomplot

timemaps=read.csv("../DATA/DATA_FILES/ddpcr-sample-time-mappings.csv",header=T) %>% select(sample.,months,leukgen_PID) 
#get the mappings of the ddpcr sample id to the absolute timepoints in months
ddpcr_full=merge(timemaps,ddpcr_outfile,by.x='sample.',by.y='Sample.ID') #merge with ddpcr results
```

```{r s3a_ddpcr_zoomplots}
theme_set(theme_bw())

plotlist=list()
for(v in pvars){
  zoomvafs=ddpcr_full %>% filter(Timepoint>0, Assay==v) %>% select(Ratio...Mu...WT..)
  titl=paste0(ddpcr_full[ddpcr_full$Assay==v,"leukgen_PID"],": ",substr(ddpcr_full[ddpcr_full$Assay==v,"Assay"],1,10))
  maxzoom=max(zoomvafs$Ratio...Mu...WT..)/100
  plot=ddpcr_full %>% filter(Timepoint>0, Assay==v) %>%
  ggplot(aes(months, Ratio...Mu...WT..)) + geom_point(color=colslist[which(pvars==v)]) + geom_line(color=colslist[which(pvars==v)]) + 
    facet_zoom(ylim=c(0,ymaxs[which(pvars==v)]),zoom.size=0.5,show.area=F, horizontal=FALSE) + 
    labs(y="VAF", title=titl) + theme(legend.position = "none", plot.title=element_text(size=10),axis.text=element_text(size=10))
  plotlist[[v]]=plot
}

plist=list(plotlist[["TP53_A276P"]],plotlist[["TP53_G266R"]],plotlist[["TP53_R175H"]],plotlist[["TP53_R249S"]],plotlist[["TP53_V172F"]],plotlist[["TP53_L257Q"]],plotlist[["TP53_R213P"]],plotlist[["TP53_R249S.2?"]])

shade_therapy=function(plot,trtfile,var){ #function to shade therapy over existing p53 plot
    for(i in 1:nrow(trtfile)) {
        for(therapy in c("chemo","RT")){
        if(!is.na(trtfile[i, paste0(therapy,".TTS")])){
          x0=trtfile[i,paste0(therapy,".TTS")]
          x1=trtfile[i,paste0(therapy,".TTE")]
          plot=plot+annotate("rect",xmin=x0/30.42,xmax=x1/30.42,ymin=0,ymax=0.55, fill="slategray",alpha=0.18)
        }
        else{
          plot=plot
        }
    }
    }
    return(plot+theme(panel.grid=element_blank(),plot.title = element_text(hjust=0.5,size=8),axis.title.x = element_blank(),axis.title.y=element_blank()))
}

pid_conv <- maindf[,c("PID","Patient...y")] %>% distinct()
rownames(pid_conv) <- substr(pid_conv$PID,5,10)
 
#patients with TP53 mutations
patlist=c(10,4,31,4,4,9,9,8)

therapy_plots=list()
for(i in 1:length(plist)){
 therapy_plots[[i]]=shade_therapy(plist[[i]],trt.table[which(trt.table$patient==patlist[i]),],i)
}

a <- arrangeGrob(therapy_plots[[1]],therapy_plots[[2]],therapy_plots[[4]],therapy_plots[[5]],therapy_plots[[3]],therapy_plots[[6]],therapy_plots[[7]],therapy_plots[[8]],ncol=1)

b=annotate_figure(a,left=text_grob("VAF",size=11,rot=90),bottom=text_grob("Time (months)", size=11))

ggsave("figs3a_ddpcr.pdf", b, device="pdf", path="../FIGURES/", width=7, height=10, useDingbats=FALSE)
```

```{r s3b_p53_growth_under_therapy}
appearance_times=DF_FIGS %>% filter(TARGET_VAF>0.01) %>% group_by(PID) %>% summarise(earliest=min(sample.time..mo.))

alphas=readRDS("../DATA/windows.rds") ##alphas, sample times and vafs for all gene level variants
dd_win=readRDS("../DATA/ddpcr_windows.rds") ##ddpcr
treat=readRDS("../DATA/treatment_intervals_all_6m.rds") %>% group_by(patient) %>% 
  mutate(cum_chemo=cumsum(chemo.days),cum_rt=cumsum(rt.events),cum_immuno=cumsum(immuno.days)) 

treat$rt_6m=lag(treat$rt_6m)
treat$chemo_6m=lag(treat$chemo_6m)
treat$immuno_6m=lag(treat$immuno_6m)

#growth rate with earliest appearance times, filtered
alpha_plus=merge(alphas,appearance_times,by='PID') %>% filter(sample.time..mo.>=earliest) 

#merge growth rates with treatment durations
main_treat=merge(alpha_plus,treat,by.x=c('Patient...x','sample_number'),by.y=c('patient','inter.num')) %>% 
mutate(treat.status=ifelse(chemo.days+immuno.days+rt.events==0,"FALSE","TRUE"),p53.status=ifelse(str_detect(VARIANT,"TP53"),"TRUE","FALSE"),ddr.status=ifelse(str_detect(VARIANT,"TP53")|str_detect(VARIANT,"PPM1D"),"TRUE","FALSE")) %>%
mutate(chemo_imm=ifelse(chemo.days*immuno.days==0,ifelse(chemo.days+immuno.days==0,"none",ifelse(chemo.days>0,"chemo","immuno")),ifelse(chemo.days/immuno.days>=5,"chemo",ifelse(chemo.days/immuno.days<=0.2,"immuno","chemo")))) %>%
  mutate(treat.combo=ifelse(rt.events==0,chemo_imm,ifelse(chemo_imm %in% c("none","immuno"),"RT",paste0(chemo_imm,"+RT"))))

main_treat=main_treat %>%
mutate(chemo_imm_6m=ifelse(chemo_6m*immuno_6m==0,ifelse(chemo_6m+immuno_6m==0,"none",ifelse(chemo_6m>0,"chemo","immuno")),ifelse(chemo_6m/immuno_6m>=5,"chemo",ifelse(chemo_6m/immuno_6m<=0.2,"immuno","chemo")))) %>%
  mutate(treat.combo.6m=ifelse(rt_6m==0,chemo_imm_6m,ifelse(chemo_imm_6m %in% c("none","immuno"),"RT",paste0(chemo_imm_6m,"+RT"))))

main_treat$treat.combo=factor(main_treat$treat.combo,levels=rev(c("chemo+RT","chemo","RT","immuno","none")))
main_treat$treat.combo.6m=factor(main_treat$treat.combo,levels=rev(c("chemo+RT","chemo","RT","immuno","none")))

main_treat=merge(alphas,treat,by.x=c('Patient...x','sample_number'),by.y=c('patient','inter.num')) %>% 
mutate(treat.status=ifelse(chemo.days+immuno.days +rt.events==0,"FALSE","TRUE"),p53.status=ifelse(str_detect(VARIANT,"TP53"),"TRUE","FALSE"),ddr.status=ifelse(str_detect(VARIANT,"TP53")|str_detect(VARIANT,"PPM1D"),"TRUE","FALSE")) %>%
mutate(treat.combo=ifelse(chemo.days==0 & rt.events>0 & immuno.days>0,"rt+immuno",ifelse(chemo.days>0 & rt.events==0 & immuno.days>0,"chemo+immuno",ifelse(chemo.days>0 & rt.events>0 & immuno.days==0,"chemo+rt",ifelse(chemo.days>0 & rt.events>0 & immuno.days>0,"chemo+rt+immuno",ifelse(chemo.days>0 & rt.events==0 & immuno.days==0,"chemo",ifelse(chemo.days==0 & rt.events>0 & immuno.days==0,"rt",ifelse(chemo.days==0 & rt.events==0 & immuno.days>0,"immuno","none"))))))))

main_treat$rt_6m=lag(main_treat$rt_6m)
main_treat$chemo_6m=lag(main_treat$chemo_6m)
main_treat$immuno_6m=lag(main_treat$immuno_6m)

main_treat=main_treat %>% 
mutate(treat.combo.6m=ifelse(chemo_6m==0 & rt_6m>0 & immuno_6m>0,"rt+immuno",ifelse(chemo_6m>0 & rt_6m==0 & immuno_6m>0,"chemo+immuno",
  ifelse(chemo_6m>0 & rt_6m>0 & immuno_6m==0,"chemo",ifelse(chemo_6m>0 & rt_6m>0 & immuno_6m>0,"chemo+rt+immuno", ifelse(chemo_6m>0 & rt_6m==0 & immuno_6m==0,"chemo",
  ifelse(chemo_6m==0 & rt_6m>0 & immuno_6m==0,"rt",ifelse(chemo_6m==0 & rt_6m==0 & immuno_6m>0,"immuno","none"))))))))

main_clean=main_treat %>% filter(!sample_number==1,ddr.status==T,cum_immuno>=0)

simple.fit=lm(formula=alpha ~ treat.combo + p53.status, data=main_clean)

summary(simple.fit)

main_clean %>% filter(cum_immuno>0) %>% ggplot(aes(x=chemo.days,y=alpha)) + geom_point() + geom_smooth(method='lm')

theme_set(theme_minimal())
main_clean %>% filter(cum_immuno>0,p53.status==T) %>% rowwise() %>% mutate(treat.status=ifelse(treat.combo=="immuno",'FALSE',treat.status)) %>% 
  ggplot(aes(x=treat.status,y=alpha,fill=treat.status,color=treat.status)) + geom_point() + geom_violin(alpha=0.75) + 
    stat_compare_means(method='t.test', size=3.5, label.x=0.9, label.y=6.8) + scale_fill_manual(values=c("thistle4","lightsteelblue"),labels=c("none/\nimmunotherapy", "RT/\nchemotherapy")) + 
    scale_color_manual(values=c("thistle4","lightsteelblue"), labels=c("none/\nimmunotherapy", "RT/\nchemotherapy")) + 
    scale_x_discrete(labels=c("none/\nimmunotherapy", "RT/\nchemotherapy"))+ labs(title="TP53 variant growth under therapy", x="Treatment Status", y="Growth Rate") +
    theme(text=element_text(size=9), title=element_text(size=10), axis.text=element_text(size=), plot.title=element_text(hjust=0.5), legend.position="none")

ggsave(file="figs3b_p53_growth.pdf",path="../FIGURES", width=4, height=5, useDingbats=FALSE)
```

```{r s4a_exponential}
patcount=DF_FIGS %>% group_by(patient.category.2, sample_number) %>% summarise(total=length(PID), pats=length(unique(PID))) 
#get number of patients that have a particular sample number/timepoint

muts=ALL_RECORDS %>% filter(TARGET_VAF>=0) #includes unknowns, every non-zero variant

trans_filt=ALL_RECORDS %>% filter(patient.category.2=="transformation") %>% mutate(timeid=paste0(sample.time..mo.,PID)) %>% select(sample.time.before.transf,timeid)

mutsfilt=muts %>% filter(patient.category.2=="transformation", VAR_TYPE %in% c("sub","indel")) %>% mutate(timeid=paste0(sample.time..mo.,PID)) %>% select(VARIANT,timeid, PID, TARGET_VAF,VAR_TYPE)

ttt=merge(trans_filt, mutsfilt) %>% unique()

#a barplot of mutation frequency approaching tmn
ttt %>% filter(sample.time.before.transf>=0,TARGET_VAF>0) %>%
ggplot(aes(x=-sample.time.before.transf)) + 
  geom_histogram(bins=8,alpha=0.55,color="gray50",fill="lightcyan3") + 
  labs(x="time pre tMN", y="variant count", title="mutation frquency approaching tMN") + 
  theme(plot.title = element_text(hjust=0.5,size=7))

#max vaf of all mutations at a timpoint across all premalignant and tmn samples for tmn patient group
ttsum=ttt %>% group_by(PID,sample.time.before.transf) %>% summarise(av=max(TARGET_VAF)) 
ttsum$sample.time.before.transf[which(is.na(ttsum$sample.time.before.transf))]=0
ggplot(ttsum, aes(x=-sample.time.before.transf, y=av)) + 
  geom_point(color="gray40",size=1.8,alpha=0.7) + geom_point(data=subset(ttsum,-sample.time.before.transf>=-3),color="darkorange",size=1.8,alpha=0.8) + 
  geom_smooth(se=F, color="lightsteelblue") + labs(x="months pre tMN", y="max VAF", title = "max VAF approaching tMN") + 
  theme(plot.title = element_text(hjust=0.5, size=8)) + scale_x_continuous(breaks=c(0,-24,-48,-72,-96,-120,-144,-168,-192,-216))

ggsave("figs4a_exponential.pdf",device="pdf", path="../FIGURES/",width=7,height=6, useDingbats=FALSE)
```

```{r s4b_allheatmaps}
## nt is generated in code for main Figure 2A in "main_figures.Rmd"

D=nt %>% ggplot(aes(x=sample_number, y=patient, fill=log(vaf))) + 
  geom_tile(color="black") + facet_wrap(~patient.category.2, scales="free_y")  +
  labs(title="Mutational timeline: GMs", x="sample number" , y="patient ID", fill="log(max VAF)") + 
  theme(plot.title=element_text(size=8, hjust = 0.5), axis.text.y=element_text(size=7, margin=margin(r=3)), 
    axis.text.x=element_text(size=7, margin=margin(t=6)), axis.title=element_text(size=8), legend.position = "bottom", legend.title = element_text(size=8, vjust=0.75), 
    legend.key.size = unit(0.5,"cm"),legend.text=element_text(size=7), strip.text = element_text(size=7), panel.spacing.x=unit(0.5,"cm")) + 
  scale_fill_gradient2(low="gray95",mid="lightsteelblue",high="darkorange",na.value = "white",midpoint=-2.6) + scale_x_continuous(expand=c(0.01,0), breaks=c(2,4,6,8))

D

ggsave("figs4b_allheatmapsGM.pdf",device="pdf", path="../FIGURES/",width=8,height=5, useDingbats=FALSE)
```

s5 is timelines

```{r s6_impact_transient_violin}
impact.vafs <- readRDS("../DATA/IMPACT_VAFS.rds")
ch.peds.all <- impact.vafs[,3:ncol(impact.vafs)]
chp.melt <- reshape2::melt(ch.peds.all)
chp.melt <- chp.melt[chp.melt$value>0,]
impact.ch.vafs <- chp.melt$value 
ch.peds.mypd <- impact.vafs[impact.vafs$ch_my_pd==1,3:ncol(impact.vafs)]
chp.mypd.melt <- reshape2::melt(ch.peds.mypd)
chp.mypd.melt <- chp.mypd.melt[chp.mypd.melt$value>0,]
impact.chpd.vafs <- chp.mypd.melt$value

df <- readRDS("../DATA/DF_FIGS.rds")
df %>% filter(VAR_TYPE %in% c("sub", "indel") & ANNOTATION %in% c("somatic | oncogenic", "somatic | likely") & sample.timepoint.2=="tMN" & TARGET_VAF>=0.02 & RM_VAR=="no") %>%
  select(TARGET_VAF) %>% unlist() %>% as.numeric() -> tmn.vaf
df %>% filter(VAR_TYPE %in% c("sub", "indel") & ANNOTATION %in% c("somatic | oncogenic", "somatic | likely") & sample.timepoint.2=="premalignant" & TARGET_VAF>=0.02 & RM_VAR=="no") %>% 
  group_by(PID,VARIANT) %>% filter(rank(sample.time..mo.)==1) %>% ungroup() %>% select(TARGET_VAF) %>% unlist() %>% as.numeric() -> premalig.vaf
df %>% filter(VAR_TYPE %in% c("sub", "indel")  & ANNOTATION %in% c("somatic | oncogenic", "somatic | likely") & sample.timepoint.2=="transient" & TARGET_VAF>0.02 & RM_VAR=="no") %>% 
  select(TARGET_VAF) %>% unlist() %>% as.numeric() -> trans.vaf

ch.vaf <- chp.melt$value
ch.mypd.vaf <- chp.mypd.melt$value
vaf.allgrps <- rbind(data.frame(VAF=impact.chpd.vafs,grp="CH"), rbind(data.frame(VAF=premalig.vaf,grp="premalignant"), data.frame(VAF=trans.vaf,grp="transient")))
theme_set(theme_bw())
ggplot(vaf.allgrps, aes(x=grp, y=VAF,color=grp)) + geom_violin(trim=FALSE,size=0.7) + stat_summary(fun.data=mean_sdl, geom="pointrange", color="gray60") + 
  scale_color_manual(values=c("lightsteelblue","lightsalmon","olivedrab3")) + labs(x="group",color="group") + 
  stat_compare_means(comparisons=list(c("CH","premalignant"),c("premalignant","transient"),c("CH","transient")),label="p.signif",hide.ns = FALSE,bracket.size = 0.15, size=2.8) +
  theme(text=element_text(size=7), title=element_text(size=8), legend.position="none")

ggsave(filename="figs6_impact_violin.pdf", path="../FIGURES/", device="pdf", width=3.5, height=4, useDingbats=FALSE)
```

```{r s7_CH-pd_piechart_older}
iold.plot <- readRDS("../DATA/IMPACT_OLDER.rds")

iold.plot$ch_final <- ifelse(iold.plot$ch_my_pd, "CH-PD", ifelse(iold.plot$CH_nonsilent, "CH non-PD", "No CH"))
iold.plot$ch_final <- factor(iold.plot$ch_final, levels=rev(c("CH-PD", "CH non-PD", "No CH")))

nold <- nrow(iold.plot)
iold.tbl.ch <- table(iold.plot$ch_final)
tbl.prop.old <- iold.tbl.ch/nold

ch.prop.old <- data.frame(t(rbind(iold.tbl.ch,tbl.prop.old)))
colnames(ch.prop.old) <- c("num", "prop")
ch.prop.old$categ <- rownames(ch.prop.old)

ch.prop.old <- ch.prop.old %>%
  arrange(desc(categ)) %>%
  mutate(lab.ypos = cumsum(prop) - 0.5*prop)
ggplot(ch.prop.old, aes(x = "", y = prop, fill = categ)) +
  geom_bar(width = 1, stat = "identity") +
  coord_polar("y", start = 0)+
  geom_text(aes(y = lab.ypos, label = round(prop,2)))+
  scale_fill_manual(values = c("powderblue", "sienna2","#6b517b")) +
  theme_void()

ggsave(filename="figs7b_impact_piechart_OLDER_RAW.pdf", path="../FIGURES/", device="pdf", useDingbats=FALSE)


```